code/build/android-native/java/org/orx/lib/OrxNativeActivity.java

```diff
@@ -0,0 +1,139 @@
+package org.orx.lib;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.app.NativeActivity;
+import android.content.Context;
+import android.hardware.input.InputManager;
+import android.view.InputDevice;
+
+/**
+    Base class for the apps NativeActivity adapted to orxAndroidNativeSupport.cpp in orxlib.
+
+    NativeActivity is used with the android_native_app_glue.
+ */
+public class OrxNativeActivity extends NativeActivity implements InputManager.InputDeviceListener {
+
+    public static int orxANDROID_KU32_MAX_JOYSTICK_NUMBER = 16; // same as in Orx /code/include/io/orxJoystick.h
+
+    // C functions we call
+    public native void nativeOnInputDeviceAdded(int deviceId);
+    public native void nativeOnInputDeviceChanged(int deviceId);
+    public native void nativeOnInputDeviceRemoved(int deviceId);
+    private native void nativeOnPause();
+    private native void nativeOnResume();
+
+    private InputManager mInputManager;
+    private int mGameControllerIds[] = new int[orxANDROID_KU32_MAX_JOYSTICK_NUMBER];
+
+    /*
+      NOTE: Do NOT to call any native methods before App is created (before this callback is done).
+      The native app structure is initialized in new thread when super.onCreate() is being executed.
+      So best is to wait until onStart() below has been called.
+    */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mInputManager = (InputManager)this.getSystemService(Context.INPUT_SERVICE);
+    }
+
+    @Override
+    protected void onStart() {
+      Log.d("OrxNativeActivity", "onStart()");
+    	super.onStart();
+
+      getAndStoreDeviceIds();
+      mInputManager.registerInputDeviceListener(this, null);

```

Comment:
Shouldn't this call be in `onResume()`?

Reply:
Actually, I refactored so we can skip to keep the device IDs in the activity. I'll push an update.
The call to register- and unregisterInputDeviceListener I just adopted from the old code OrxActivity onStart()/onStop() and I it seems to work well when I tested myself.

Reply:
Activity life cycle management is usually hard to test thoroughly. I believe `onPause` and `onResume` are more correct.

The "pause" event occurs when an activity is *preparing* to stop. E.g. it becomes semi-transparent in some exit animation. "Stop" arrives when the activity is completely invisible.

Not sure what's best here (or if it matters), but most sources on the net seem to use the pause/resume pattern.

Reply:
I did not get and PAUSE and RESUME when run last time on my Android TV box.

Reply:
for me this is good
@jonasohrn up to you to resolve the conversation